chain-abci/src/app/mod.rs

```diff
@@ -148,6 +149,27 @@ impl<T: EnclaveProxy> abci::Application for ChainNodeApp<T> {
             }
         }
 
+        let mut accounts_to_jail = Vec::new();
+
+        for evidence in req.byzantine_validators.iter() {
+            if let Some(validator) = evidence.validator.as_ref() {
+                let validator_address =
+                    TendermintValidatorAddress::try_from(validator.address.as_slice())
+                        .expect("Invalid validator address in begin block request");
+                let account_address = last_state
+                    .validator_liveness
+                    .get(&validator_address)
+                    .expect("Validator not found in liveness tracker")
+                    .address();
+                accounts_to_jail.push(account_address);
+            }
+        }
+
+        for staking_address in accounts_to_jail {
+            self.jail_account(staking_address)
```

Comment:
this could be moved up to L164 instead of building the extra intermediate vector